src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -508,7 +508,7 @@ class Namer { typer: Typer =>
         if (sym is Module) moduleValSig(sym)
         else valOrDefDefSig(original, sym, Nil, Nil, identity)(localContext(sym).setNewScope)
       case original: DefDef =>
-        val typer1 = new Typer
```

Comment:
@odersky, I'm curious why getting a signature of a DefDef requires creating a new Typer\Namer, while ValDefs and TypeDefs are fine using existing one?